src/deluge/memory/cache_manager.cpp

```diff
@@ -0,0 +1,176 @@
+#include "cache_manager.h"
+#include "io/debug/print.h"
+#include "memory/memory_region.h"
+#include "memory/stealable.h"
+#include "processing/engines/audio_engine.h"
+#include <tuple>
+
+extern bool skipConsistencyCheck;
+uint32_t currentTraversalNo = 0;
+
+// Size 0 means don't care, just get any memory.
+uint32_t CacheManager::ReclaimMemory(MemoryRegion& region, int totalSizeNeeded, void* thingNotToStealFrom,
+                                     int* __restrict__ foundSpaceSize) {
+
+#if TEST_GENERAL_MEMORY_ALLOCATION
```

Comment:
Have you tried running with this flag? No idea what it is

Reply:
Nope, it theoretically tests the memory but I have no idea if the code in the "tests" are still valid for this version.